src/v/cloud_io/io_resources.cc

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "cloud_io/io_resources.h"
+
+#include "base/vlog.h"
+#include "cloud_io/logger.h"
+#include "config/configuration.h"
+#include "config/node_config.h"
+#include "resource_mgmt/io_priority.h"
+#include "ssx/future-util.h"
+#include "utils/token_bucket.h"
+
+#include <seastar/core/io_queue.hh>
+#include <seastar/core/reactor.hh>
+
+namespace cloud_io {
+
+namespace {
+
+class throttled_dl_source : public ss::data_source_impl {
```

Comment:
> cloud_io: separate out materialized_resources

is this new code? the commit message makes it seem like it would be code that is moved

Reply:
Yea i should probably restructure this PR a bit. This is taken out of cloud_storage::materialized_resources

Oh oops actually I forgot to remove the code in materialized_resources